sdk/src/main/java/io/customer/sdk/CustomerIO.kt

```diff
@@ -0,0 +1,128 @@
+package io.customer.sdk
+
+import android.content.Context
+import io.customer.base.comunication.Action
+import io.customer.sdk.api.CustomerIoApi
+import io.customer.sdk.data.model.Region
+import io.customer.sdk.di.CustomerIOComponent
+
+/**
+Welcome to the Customer.io Android SDK!
+
+This class is where you begin to use the SDK.
+You must have an instance of `CustomerIO` to use the features of the SDK.
+
+Create your own instance using
+`CustomerIo.Builder(siteId: "XXX", apiKey: "XXX", region: Region.US, appContext: Application context)`
+It is recommended to initialize the client in the `Application::onCreate()` method.
+
+After the instance is created you can access it via singleton instance: `CustomerIO.instance()` anywhere,
+ */
+
+class CustomerIO internal constructor(
+    val config: CustomerIOConfig,
+    private val api: CustomerIoApi,
```

Comment:
`CustomerIO.Companioninstance` creates a singleton instance of the `CustomerIO` class. 

Looking at all of the properties of the `CustomerIO` class, `CustomerIOConfig` at this time is safe to keep a strong reference of as it's just a collection of Strings and such. 

I am, however, nervous about keeping a strong reference to `CustomerIoApi` in this class as that class has dependencies which those have dependencies, etc, etc. It seems prone to a memory leak bug in the future. 

Any suggestions for how we can prevent in the code base? 

...I am glad I am seeing this now because [I did something similar on iOS](https://github.com/customerio/customerio-ios/blob/c0eb2737bc2727e7aa64ce1e43b33447766e943a/Sources/Tracking/CustomerIOImplementation.swift#L26) that should be addressed!

Reply:
I am not necessarily suggesting we change the way we do dependency injection. You using constructor injection is a good thing. 

I am curious if there is a safer way to manage the `CustomerIO` singleton class. It's the fact that the singleton class has a strong reference to a dependency (in this case a Repository) and the Repository has many nested dependencies inside of it is what I mean by this being error-prone. If we were able to not make `CustomerIO` class a singleton instance (not likely?) or try to limit the strong references in the class to instances that are light-weight (not having a lot of nested dependencies could fix this issue. 

This is more of a conversation. Not something to fix in this code review.